components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java

```diff
@@ -1238,18 +1238,24 @@ public static String getApplicationUpdatedVersion(ServiceProvider serviceProvide
         String currentVersion = serviceProvider.getApplicationVersion();
         String inboundConfigType = getInboundConfigType(serviceProvider);
 
-        // Since there will be new versions onboarded in the future, initialized this as a switch not if.
-        if (currentVersion.equals(ApplicationConstants.ApplicationVersion.APP_VERSION_V0)) {
-            if (!inboundConfigType.equals(IdentityApplicationConstants.OAuth2.NAME)) {
-                currentVersion = ApplicationConstants.ApplicationVersion.APP_VERSION_V1;
-            }
+        switch (currentVersion) {
+            case ApplicationConstants.ApplicationVersion.APP_VERSION_V0:
+            case ApplicationConstants.ApplicationVersion.APP_VERSION_V1:
+                if (inboundConfigType != null && !inboundConfigType.equals(IdentityApplicationConstants.OAuth2.NAME)) {
```

Comment:
Swap the constant with the variable

Reply:
The null check was there for apps with no inbound auth method configured. But we can change that logic to behave as same as non-oidc app. There for changing the code as below.

834a75d5812a461ce8d80e2386d84292e986fbb7